sys/net/gnrc/transport_layer/udp/gnrc_udp.c

```diff
@@ -130,9 +130,9 @@ static void _receive(gnrc_pktsnip_t *pkt)
     port = (uint32_t)byteorder_ntohs(hdr->dst_port);
 
     /* send payload to receivers */
-    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, pkt)) {
+    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, udp)) {
```

Comment:
it seems to be just this line that breaks the current state. But I don't think this is really a solution here... I think you intented to send the **payload** to the listener and not the udp snip itself. This is what I changed here now. pktdump confirms, it begins printing with the udp snip instead of the payload. So the actual problem may be somewhere else .. maybe in pktdump?